source/upgrade/prepare-to-upgrade-mattermost.rst

```diff
@@ -0,0 +1,187 @@
+Prepare to upgrade Mattermost
+=============================
+
+|all-plans| |self-hosted|
+
+.. |all-plans| image:: ../images/all-plans-badge.png
+  :scale: 30
+  :target: https://mattermost.com/pricing
+  :alt: Available in Mattermost Free and Starter subscription plans.
+
+.. |self-hosted| image:: ../images/self-hosted-badge.png
+  :scale: 30
+  :target: https://mattermost.com/deploy
+  :alt: Available for Mattermost Self-Hosted deployments.
+
+In most cases, you can upgrade Mattermost Server in a few minutes. However, the upgrade can take longer depending on several factors, including the size and complexity of your installation, and the version that you're upgrading from.
+
+.. contents::
+  :backlinks: top
+  :local:
+
+Upgrade to Mattermost v7.1
+--------------------------
+
+Mattermost v7.1 introduces schema changes in the form of a new column and its index. The following notes our test results for the schema changes:
+
+- MySQL 12M Posts, 2.5M Reactions - ~1min 34s (instance: PC with 8 cores, 16GB RAM)
+- PostgreSQL 12M Posts, 2.5M Reactions - ~1min 18s (instance: db.r5.2xlarge)
+
+Customers wanting an upgrade with substantially reduced downtime (~1/5th of the original time) are encouraged to run the following queries before upgrading. This is fully backwards-compatible, but will obtain a lock on ``Reactions`` table, so users' reactions posted during this time won't be reflected in the database.
```

Comment:
What are the impacts of the lock and not posting the reactions when running the queries not in a maintenance window?

Reply:
@agnivade - Can you help answer Neil's question?

Reply:
@nab-77 - The impact is mentioned - "so users' reactions posted during this time won't be reflected in the database." Let me know if you have any more questions.

This is just like any other schema migration change that we do every release. Some migrations will lock the tables, some won't. When a migration locks a table, nothing gets written to that table during that time.

cc @plant99

Reply:
@nab-77 - Based on Agniva's input, I've also updated the introductory sentence to remove the specifics about reduced downtime.

Reply:
@nab-77 The data isn't lost because it's not accepted in the first place, it'll be rejected by the database which you'll see reflected in the API response.